[quant][graphmode] Add support for quantized conv1d + relu fusion

supriyar opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #38452 [quant] Support for fused ConvBn1d and ConvBnRelu1d modules
* #38449 [quant] Support for functional quantized::conv1d
* **#38441 [quant][graphmode] Add support for quantized conv1d + relu fusion**
* #38438 [quant] Fusion support for conv1d + ReLU
* #38407 [quant] Return default qconfig when backend is 'none'

Summary:

Test Plan:
python test/test_quantization.py test_quantized_conv1d_relu

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D21575919](https://our.internmc.facebook.com/intern/diff/D21575919)

supriyar committed:
[quant][graphmode] Add support for quantized conv1d + relu fusion
Summary:

Test Plan:
python test/test_quantization.py test_quantized_conv1d_relu

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

raghuramank100 commented on torch/csrc/jit/passes/quantization/quantization_patterns.h:
```diff
@@ -32,6 +32,40 @@ graph(%a_quant, %packed_params, %r_scale, %r_zero_point, %r_dtype, %stride, %pad
+  // aten::conv1d - aten::relu
+  std::string conv1d_relu = R"(
+graph(%a_quant, %packed_params, %r_scale, %r_zero_point, %r_dtype, %stride, %padding, %dilation, %groups):
+        %a_dequant = aten::dequantize(%a_quant)
```
For my understanding, where is the pack op inserted for the weights? I was expecting to see a wt->q->pack->unpack->dq->Conv

jerryzh168 replied:
yes, it's inserted before, here we just fuse `unpack - dq - conv - q` to quantized::conv2d

jerryzh168 commented on torch/csrc/jit/passes/quantization/insert_observers.cpp:
```diff
@@ -384,6 +384,20 @@ class InsertObserversHelper {
   // These are the IR patterns we match to skip inserting observers.
   // They are compiled once on construction and used repeatedly within
   // the pass.
+  const PatternInfo conv1d_functional_relu = PatternInfo::parse_from_str(R"(
```
let's use naming convention suggested by @z-a-f : `nn_conv1d_f_relu`, similar for others

supriyar replied:
Sure

supriyar committed:
Update on "[quant][graphmode] Add support for quantized conv1d + relu fusion"


Summary:

Test Plan:
python test/test_quantization.py test_quantized_conv1d_relu

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

supriyar committed:
Update on "[quant][graphmode] Add support for quantized conv1d + relu fusion"


Summary:

Test Plan:
python test/test_quantization.py test_quantized_conv1d_relu

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]

supriyar committed:
Update on "[quant][graphmode] Add support for quantized conv1d + relu fusion"


Summary:

Test Plan:
python test/test_quantization.py test_quantized_conv1d_relu

Reviewers:

Subscribers:

Tasks:

Tags:

[ghstack-poisoned]
